common.lic

```diff
@@ -382,6 +390,22 @@ module DRC
     end
   end
 
+  # Looks up the canonical name of the town based on the given text.
+  # Utility to help identify the canonical town name based on arbitrary text.
+  # For example, "Theren" for "Therenborough" and "Haven" for "Riverhaven".
+  # It also handles missing apostrophes and the occasional space between names
+  # like "merkresh" or "Mer'Kresh" or "ainghazal" or "Ain Ghazal".
+  # Returns nil if unable to find a match.
+  def get_town_name(text)
+    towns = $HOMETOWN_REGEX.select { |town, regex| regex =~ text }.keys
+    if towns.length > 1
+      DRC.message("Found multiple towns that match '#{text}': #{towns}")
+      DRC.message("Using first town that matched: #{towns.first}")
+      DRC.message("To avoid ambiguity, please use the town's full name: https://elanthipedia.play.net/Category:Cities")
+    end
+    towns.first
+  end
```

Comment:
In response to https://github.com/rpherbig/dr-scripts/pull/5256#discussion_r758321544

When more than one town is found given the search text then a warning is given with advice to use the town's full name, then returns the first match.

_The following test had two towns that would match 'cross'_
```
,e echo DRC.get_town_name('cross')

--- Lich: exec1 active.

| Found multiple towns that match 'cross': ["Crossing", "Crossroads"]
| Using first town that matched: Crossing
| To avoid ambiguity, please use the town's full name: https://elanthipedia.play.net/Category:Cities

[exec1: Crossing]

--- Lich: exec1 has exited.
```